rca/people/models.py

```diff
@@ -1,65 +1,172 @@
 from modelcluster.fields import ParentalKey
-from wagtail.admin.edit_handlers import FieldPanel, InlinePanel, MultiFieldPanel
+from wagtail.admin.edit_handlers import (
+    FieldPanel,
+    InlinePanel,
+    MultiFieldPanel,
+    ObjectList,
+    PageChooserPanel,
+    StreamFieldPanel,
+    TabbedInterface,
+)
+from wagtail.core.fields import RichTextField, StreamField
 from wagtail.images.edit_handlers import ImageChooserPanel
 
-from rca.utils.models import BasePage
+from rca.utils.blocks import AccordionBlockWithTitle, GalleryBlock, LinkBlock
+from rca.utils.models import BasePage, RelatedPage
 
 
-class SocialMediaProfile(models.Model):
-    person_page = ParentalKey("StaffPage", related_name="social_media_profile")
-    site_titles = (("twitter", "Twitter"), ("linkedin", "LinkedIn"))
-    site_urls = (
-        ("twitter", "https://twitter.com/"),
-        ("linkedin", "https://www.linkedin.com/in/"),
+class AreaOfExpertise(models.Model):
+    title = models.CharField(max_length=128)
+
+    def __str__(self):
+        return self.title
+
+
+class StaffPageAreOfExpertisePlacement(models.Model):
+    page = ParentalKey("StaffPage", related_name="related_area_of_expertise")
+    area_of_expertise = models.ForeignKey(
+        AreaOfExpertise, on_delete=models.CASCADE, related_name="staff"
     )
-    service = models.CharField(max_length=200, choices=site_titles)
-    username = models.CharField(max_length=255)
+    panels = [FieldPanel("area_of_expertise")]
+
 
-    @property
-    def profile_url(self):
-        return dict(self.site_urls)[self.service] + self.username
+class StaffPageRelatedResearchPage(RelatedPage):
+    source_page = ParentalKey("StaffPage", related_name="related_research_centre_pages")
+    panels = [PageChooserPanel("page", "research.ResearchCentrePage")]
 
-    def clean(self):
-        if self.service == "twitter" and self.username.startswith("@"):
-            self.username = self.username[1:]
 
+class StaffPageRelatedSchoolPage(RelatedPage):
+    source_page = ParentalKey("StaffPage", related_name="related_school_pages")
+    panels = [PageChooserPanel("page", "schools.SchoolPage")]
 
-class StaffPagePhoneNumber(models.Model):
-    page = ParentalKey("StaffPage", related_name="phone_numbers")
-    phone_number = models.CharField(max_length=255)
 
-    panels = [FieldPanel("phone_number")]
+class StaffPagePageRelatedProjects(RelatedPage):
+    source_page = ParentalKey("StaffPage", related_name="related_projects")
+    panels = [PageChooserPanel("page", "projects.ProjectPage")]
 
 
 class StaffPage(BasePage):
     template = "patterns/pages/staff/staff_detail.html"
-
+    staff_title = models.CharField(
+        max_length=255, help_text=_("E.G Dr, Professor"), blank=True
+    )
     first_name = models.CharField(max_length=255)
     last_name = models.CharField(max_length=255)
-    image = models.ForeignKey(
+    profile_image = models.ForeignKey(
         "images.CustomImage",
```

Comment:
For portability, it may be better to use `wagtail.images.get_image_model_string()` here instead of harcoding the model name

Reply:
fixed in https://github.com/torchbox/rca-wagtail-2019/pull/426/commits/88b0b258220d5a421b8f5794d9ab35fc975fee2c